apps/dashboard/src/pages/my-groups.tsx

```diff
@@ -41,26 +40,23 @@ export default function MyGroups(): JSX.Element {
 
     useEffect(() => {
         ;(async () => {
+            // If metamask is already connected, assume on-chain group
+            if (account) {
+                setIsOffchainGroup(false)
+                return
+            }
+
+            // Set as off-chain group is user logged in via SSP
             await request(`${environment.apiUrl}/auth/getUser`)
                 .then((res) => {
-                    setJwtInCookies(true)
+                    setIsOffchainGroup(true)
                 })
                 .catch((e) => {
-                    console.log("no jwt")
-                    setJwtInCookies(false)
+                    // Failure - Redirect to login page
+                    navigate("/sso")
                 })
         })()
-    }, [])
-
-    useEffect(() => {
-        if (jwtInCookies) {
```

Comment:
Not sure how it was working - when the page loads it send a request to `getUser` and `jwtInCookies  = true`. But before this happen the other `useEffect` hook check for `jwtInCookies`, which would be false and redirect the user back to `/sso`
Fixed this my merging both hooks in to one.